Modules/_datetimemodule.c

```diff
@@ -2284,21 +2284,26 @@ delta_bool(PyDateTime_Delta *self)
 static PyObject *
 delta_repr(PyDateTime_Delta *self)
 {
-    if (GET_TD_MICROSECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self),
-                                    GET_TD_MICROSECONDS(self));
-    if (GET_TD_SECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self));
-
-    return PyUnicode_FromFormat("%s(%d)",
+    PyObject *args = PyList_New(0);
+
+    if (GET_TD_DAYS(self) != 0) {
+        PyList_Append(args, PyUnicode_FromFormat("days=%d", GET_TD_DAYS(self)));
```

Comment:
`PyUnicode_FromFormat()` and `PyList_Append()` can fail.

The result of `PyUnicode_FromFormat()` is leaked.